app/assets/stylesheets/base/mixins.scss

```diff
@@ -491,11 +491,53 @@ ORDER: Base + typography > general layout + grid > page layout > components
   text-transform: uppercase;
 }
 
-// write CSS variables without like "v(thevariable);"
-@function v($var) {
-  @return var(--#{$var});
+// the variable mixin takes a property and variable name 
+// and an optional override to hide the fallback
+@mixin v($prop, $var, $show-fall: true) {
+  // get the property's map name(s)
+  $map-name: map-get($props, $prop);
+  $nest-name: null;
+  $nest-map-name: null;
+  $map: null;
+  $var-fall: null;
+  $var-output: null;
+  // if a list, we need to go deeper
+  @if type-of($map-name) == list {
+    $nest-name: nth($map-name, 1);
+    $nest-map-name: nth($map-name, 2);
+  }
+  // if it is a nested map
+  @if $nest-name {
+    // get the map from nested map-name
+    $map: map-get($var-maps, $nest-name);
+    // get the nested map
+    $nest-map: map-get($map, $nest-map-name);
+    // fallback value, get the var value from the nested map
+    $var-fall: map-get($nest-map, $var);
+    // our css4 variable output
+    $var-output: var(--#{$nest-name}-#{$nest-map-name}-#{$var});
+  } @else {
+    // get the map from map name
+    $map: map-get($var-maps, $map-name);
+    // fallback value, grab the variable's value from the map
+    $var-fall: map-get($map, $var);
+    // our css4 variable output
+    $var-output: var(--#{$map-name}-#{$var});
+  }
+  // if show fallback is not overridden to be null
+  @if $show-fall {
+    #{$prop}: $var-fall;
+  }
+  // css4 variable output
+  #{$prop}: $var-output;
 }
 
```

Comment:
This mixin is used to allow for css variables and provide a fallback for Internet Explorer 10 & 11 which can't use css variables. It works by passing in the css property and corresponding variable. Sass allows for lists so you can group varibales together using key value pairs. This mixin parses thoses lists and creates the css property using the Sass varibale and creates a css variable as well adding the css property to the front of the variable see the screenshot for the otuput of this mixin

Reply:
<img width="341" alt="Screenshot 2020-05-11 15 28 00" src="https://user-images.githubusercontent.com/29960336/81618368-0bc71880-939c-11ea-9a42-799bb6e31fcd.png">